docs/inventory_plugins.md

```diff
@@ -0,0 +1,197 @@
+plugins. For any given source, after the `initial_version` for plugin use
+is higher than the lowest supported Ansible version, the script can be
+removed and the logic for script credential injection can also be removed.
+
+For example, after AWX no longer supports Ansible 2.7, the script
+`awx/plugins/openstack_inventory.py` will be removed.
+
+## Changes to Expect in Imports
+
+An effort was made to keep imports working in the exact same way after
+the switchover. However, the inventory plugins are a fundamental rewrite
+and many elements of default behavior has changed. Because of that,
+a `compatibility_mode` toggle was added.
+
+In a data migration, all existing cloud sources are switched over to
+use `compatibility_mode`. New inventory sources will default to having
+this off.
+
+We recommend that you opt out of compatibility mode because this is more
+future-proof, and also suggest that you set the `overwrite`
+flag to help assure stale content is removed.
+
+### Changes with Compatibility Mode Off
+
+If no `group_by` entries are given, then no constructed groups will be
+produced. That means no grouping by tags, regions, or similar attributes
+unless the user adds these to the `group_by` listing. This is different
+from prior behavior, where a blank `group_by` field would include all
+possible groups.
+
+The set of `hostvars` will be almost completely different, using new names
+for data which is mostly the same content. You can see the jinja2 keyed_groups
+construction used in compatibility mode to help get a sense of what
+new names replace old names.
+
+In many case, the host names will change. In many cases, accurate host
+tracking will still be maintained via the host `instance_id`, but this
+is not guaranteed.
+
+Group names will be sanitized. That means that characters such as "-" will
+be replaced by underscores "\_". In some cases, this means that a large
+fraction of groups get renamed as you move from scripts to plugins.
+Sanitizing group names allows referencing them in jinja2 in playbooks
+without errors.
+
+### Changes with Compatibility Mode On
+
+Programatically-generated examples of inventory file syntax used in
+updates (with dummy data) can be found in `awx/main/tests/data/inventory/scripts`,
+these demonstrate the inventory file syntax used to restore old behavior
+from the inventory scripts.
+
+#### hostvar keys
+
+More hostvars will appear. The inventory plugins name hostvars differently
+that the contrib scripts did. To maintain backward compatibility,
+the old names are added back where they have the same meaning as a
+variable returned by the plugin. New names are not removed.
+
+Caution: if you do not have `overwrite_vars` set
+to True and you _downgrade_ the version of Ansible that an import runs in,
+this will leave some stale hostvars.
+
+Some hostvars will be lost, because of general deprecation needs.
+
+ - ec2, see https://github.com/ansible/ansible/issues/52358
```

Comment:
This needs real docs whenever it's settled, not just a link. It can go in product documentation.